src/service/ratelimit_legacy.go

```diff
@@ -0,0 +1,143 @@
+package ratelimit
```

Comment:
Why maintain this code at all? Can't you just convert to YAML and back?

Reply:
So the reason I went with this route was to potentially squeeze some perf out of manipulating only golang structs, and not having to deal with marshal/unmarshal oddities which I have had to deal with in envoymanager.

But, the approach you suggest is less code to maintain, and less surface for developer error. I pushed that approach below.